File: experimental/oo-language/src/main/scala/de/tuda/stg/consys/experimental/lang/store/Store.scala
package de.tuda.stg.consys.experimental.lang.store

/**
 * Created on 10.12.19.
 *
 * @author Mirko Köhler
 */
trait Store {

	type Addr
	type ObjType

	type Context <: TxContext

	type RefType[_ <: ObjType] <: Handler[_ <: ObjType]


	def transaction[T](code : Context => Option[T]) : Option[T]

}

File: consys-japi/src/main/java/de/tuda/stg/consys/japi/Store.java
package de.tuda.stg.consys.japi;

import scala.Option;

/**
 * Created on 27.01.20.
 *
 * @author Mirko Köhler
 */
public interface Store<Addr, Obj, Consistency, TxContext extends TransactionContext<Addr, Obj, Consistency>> {

	<U> Option<U> transaction(Transaction<TxContext, U, Addr, Obj, Consistency> tx);

	void close() throws Exception;
}
